### DISCUSSION TITLE
Legacy version of the msprime tutorial?

### DISCUSSION THREAD

**Discussion by saramathieson** [tskit-dev/msprime#1704] [Q&A]
Hi all - I was wondering if there was a legacy version of the msprime tutorial (originally here: http://msprime.readthedocs.io/en/stable/tutorial.html) available somewhere?  I don't know if there would be general interest in setting this up, but I have a few things that depend on legacy code at the moment.  Thanks!

**Comment by petrelharp** [comment]
I think currently we don't have those up anywhere, although perhaps we should - @benjeffery would be the person to ask about setting that up, and he's offline at the moment. You can at least build them locally - doing e.g. `git checkout 0.7.5 && cd docs && make` (maybe after `pip3 install requirements/CI-docs/requirements.txt`) should do it.

**Reply by jeromekelleher** [reply]
I was *just* about to write exactly the same thing @petrelharp!

**Reply by petrelharp** [reply]
lol we need an "I got this" flag

**Comment by jeromekelleher** [comment]
Does building the docs locally work for you @saramathieson? We probably should make a "legacy" version of the doc available, but we're a bit short-handed at the moment so it'll be a few weeks before it happens.

**Reply by saramathieson** [reply]
Thanks for the replies!  I installed all the libraries in `requirements/readthedocs.txt`, but unfortunately I'm running into some errors with `make`.  I think these can be traced to sphinx version issues.  Do either of you happen to know what versions of the required libraries were used at the time this documentation was built?  I also tried a few other ways to generate the html but no luck so far.

**Reply by petrelharp** [reply]
Do the versions in `requirements/CI-docs/requirements.txt` do it?

**Reply by saramathieson** [reply]
There aren't any folders in the `requirements` folder (just the files `conda-minimal.txt`, `development.txt`, `readthedocs.txt`, `tests-pypi.txt`, `verification.txt`).  This is `0.7.5`.  Maybe I'm missing something?

**Reply by jeromekelleher** [reply]
It worked for me with the current versions of the packages in ``requirements/readthedocs.txt`` after checking out 0.7.5 @saramathieson - can you paste in the error you're getting please?

**Reply by petrelharp** [reply]
Sorry - those requirements files are in the current head, not 0.7.5.

**Reply by saramathieson** [reply]
Here is the error - maybe it's an issue with `breathe` not `sphinx`?  I got the same error on two separate computers.  I think this is going beyond general `msprime` help, so I'm also happy to wait on this :)

```
$ make
sphinx-build -b html -d _build/doctrees  -W . _build/html
Running Sphinx v4.0.2
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from http://docs.scipy.org/doc/numpy/objects.inv...
loading intersphinx inventory from https://tskit.readthedocs.io/en/stable/objects.inv...
intersphinx inventory has moved: https://tskit.readthedocs.io/en/stable/objects.inv -> https://tskit.dev/tskit/docs/stable/objects.inv
intersphinx inventory has moved: http://docs.scipy.org/doc/numpy/objects.inv -> https://numpy.org/doc/stable/objects.inv
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 9 source files that are out of date
updating environment: [new config] 9 added, 0 changed, 0 removed
reading sources... [ 22%] api                                                                                             
Exception occurred:
  File "/Library/Python/3.8/site-packages/sphinx/domains/python.py", line 291, in make_xref
    result['py:module'] = env.ref_context.get('py:module')
AttributeError: 'NoneType' object has no attribute 'ref_context'
The full traceback has been saved in /var/folders/0b/5vp8tkqn3hx_p1l29hdv1zrw0000gq/T/sphinx-err-qz7uy7g4.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [html] Error 2
```

**Reply by petrelharp** [reply]
Hmph. It works for me, on Debian, with:
```
breathe=4.22.1
sphinx=3.2.1
sphinx-argparse=0.2.5
sphinx-issues=1.2.0
setuptools-scm=1.11.1
```

**Reply by jeromekelleher** [reply]
Yeah, I'm stumped here, I don't understand what's going on.

**Reply by saramathieson** [reply]
@petrelharp that worked!  I switched to linux and installed these exact versions (as well as `sphinx-rtd-theme=0.5.2` and `tskit=0.3.6`) and then everything worked perfectly.  Thanks all - I really appreciate it!

**Comment by grahamgower** [comment]
There's a fairly old version of the docs in the wayback machine. Maybe that's useful to you @saramathieson? http://web.archive.org/web/20180616223919/http://msprime.readthedocs.io/en/stable/tutorial.html

**Reply by saramathieson** [reply]
This is great to have as a non-local option - thanks!

**Comment by benjeffery** [comment]
I've added https://github.com/tskit-dev/tskit-site/issues/45 as it would seem worthwhile to have legacy docs for a version that was used for so long available.
